### PR TITLE
feat: codama interactive idl

### PR DESCRIPTION
**Pull request by askov**
## Description

This PR contains support for Codama interactive IDLs.

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Protocol integration
-   [ ] Documentation update
-   [ ] Other (please describe):

### Programs with uploaded Codama IDLs:
- Let Me Buy
    - Original `BUYuxRfhCMWavaUWxhGtPP3ksKEDZxCD5gzknk3JfAya`
    - Fork `8N7W6D2hs9rcMfgKDWKpDHSFdeUaneFSa4wtD7BRSXY`
- Voting `AXcxp15oz1L4YYtqZo6Qt6EkUj1jtLR6wXYqaJvn4oye`
- Blog `4ypHLeD2zDinCMiuHnwpRTwygKDxR5Sn6wbVHy4mbnvE`
- Example with random edge cases `5v4CbtQTxb4iYAW1MCJMpVu5Dud9ybmRqzXs4bTCJm3o`
- PMP `ProgM6JCCvbYkfKqJYHePx4xxSUSqJp7rh8Lyv7nk7S`

## Checklist

-   [ ] My code follows the project's style guidelines
-   [x] I have added tests that prove my fix/feature works
-   [x] All tests pass locally and in CI
-   [ ] I have updated documentation as needed
-   [x] I have run `build:info` script to update build information
-   [x] CI/CD checks pass
-   [ ] I have included screenshots for protocol screens (if applicable)
-   [ ] For security-related features, I have included links to related information

### CODE REVIEW

File: app/features/idl/interactive-idl/model/pda-generator/codama-provider.ts [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+        const seedName = camelCase(seed.name);
+        seedInfo.push({ name: seedName, value: formValue });
+
+        if (formValue === null) {
+            allResolved = false;
+            continue;
+        }
+
+        try {
+            seedInputs[seed.name] = convertValue(formValue, seed.type as TypeNode);
```
[the remark this answers is not in the document]
Comment (mikhd) [reply app/features/idl/interactive-idl/model/pda-generator/codama-provider.ts]: https://github.com/hoodieshq/explorer/pull/87/commits/5a5e6e0546391fe5986d3a7ccf6c71e813303896

**Review by mikhd — APPROVED** [review]

File: app/features/idl/interactive-idl/model/__mocks__/codama/codama-voting.json [hunk 2]
```diff
@@ -0,0 +1,1172 @@
+{
```
Comment (rogaldh) [inline app/features/idl/interactive-idl/model/__mocks__/codama/codama-voting.json:1]: Let's maybe minify the mocker IDL as we're not gonna edit it anyway?

File: app/features/idl/interactive-idl/model/pda-generator/types.ts [hunk 3]
```diff
@@ -52,8 +63,12 @@ export type PdaProvider = {
     getProgramId: (idl: SupportedIdl) => PublicKey | null;
 
     /**
-     * Find instruction by name in the IDL
-     * Returns null if instruction not found
+     * Compute PDA addresses for an instruction's accounts.
      */
-    findInstruction: (idl: SupportedIdl, instructionName: string) => PdaInstruction | null;
```
Comment (rogaldh) [inline app/features/idl/interactive-idl/model/pda-generator/types.ts:58]: `findInstruction` is still present at AnchorProvider. I think we should keep it

Comment (mikhd) [reply app/features/idl/interactive-idl/model/pda-generator/types.ts:58]: I think it was removed as well https://github.com/hoodieshq/explorer/pull/87/files#diff-5b1e4c557580f43e6308264bad2756f1b8aa033a1104011675072f3089a6e112

So do we actually need it?

File: app/features/idl/interactive-idl/model/__mocks__/codama/codama-voting.json (hunk 2 above)
Comment (mikhd) [reply app/features/idl/interactive-idl/model/__mocks__/codama/codama-voting.json:1]: Fixed in separate PR https://github.com/hoodieshq/explorer/pull/88
